lib/ast.js

```diff
@@ -191,17 +191,20 @@ util.methods(Symbols, {
     this.register('functions', name, fn);
   },
 
-  registerPath: function(parts, methods) {
+  registerPath: function(parts, isType, methods) {
+    isType = isType || 'Any';
     var p = {
       parts: parts,
+      isType: isType,
       methods: methods
     };
     this.register('paths', '/' + parts.join('/'), p);
   },
 
   registerSchema: function(name, derivedFrom, properties, methods) {
+    derivedFrom = derivedFrom || (Object.keys(properties).length > 0 ? 'Object' : 'Any');
```

Comment:
Seems like Any should be anything rather than anything but Object (primitives mainly?). Is there an advantage to distinguishing between Any vs. Object?